File: TransCoder_tokenized_test_set_functions/python/COUNT_SUBSTRINGS_BINARY_STRING_CONTAINING_K_ONES.py
def countOfSubstringWithKOnes ( s, K ) :
    N = len ( s )
    res = 0
    countOfOne = 0
    freq = [ 0 for i in range ( N + 1 ) ]
    freq [ 0 ] = 1
    for i in range ( 0, N, 1 ) :
        countOfOne += ord ( s [ i ] ) - ord ( '0' )
        if ( countOfOne >= K ) :
            res += freq [ countOfOne - K ]
        freq [ countOfOne ] += 1
    return res

File: TransCoder_tokenized_test_set_functions/cpp/COUNT_SUBSTRINGS_BINARY_STRING_CONTAINING_K_ONES.cpp
int countOfSubstringWithKOnes ( string s, int K ) {
  int N = s . length ( );
  int res = 0;
  int countOfOne = 0;
  int freq [ N + 1 ] = {
    0 };
    freq [ 0 ] = 1;
    for ( int i = 0;
    i < N;
    i ++ ) {
      countOfOne += ( s [ i ] - '0' );
      if ( countOfOne >= K ) {
        res += freq [ countOfOne - K ];
      }
      freq [ countOfOne ] ++;
    }
    return res;
  }